### PR TITLE
Stripe integration

### PR DESCRIPTION
**Pull request by rlconley**
This is a basic working payment that has been tested against the stripe sandbox.  It has one donation field that the user can edit and a tip field that the user can edit.  I'd especially appreciate feedback on error handling and tests that would be needed.

### CODE REVIEW

File: revolv/payments/models.py [hunk 1]
```diff
@@ -528,3 +539,12 @@ class ProjectMontlyRepaymentConfig(models.Model):
...
     def __unicode__(self):
         return '%s %s in %s for %s' % (self.repayment_type, self.amount, self.year, self.project)
+
+
+class Tip(models.Model):
+    """
+        Percentage of payment user elects to add to donation toward Revolv overhead costs
+    """
+    timestamp = models.DateTimeField(auto_now_add=True, blank=True)
+    user = models.ForeignKey('base.RevolvUserProfile')
```
Comment (trawick) [inline revolv/payments/models.py:549]: Was the migration generated when this was a ForeignKey to User?  I think that's what the new migration indicates.

File: revolv/project/views.py [hunk 2]
```diff
@@ -14,12 +15,53 @@
...
 from revolv.payments.services import PaymentService
 from revolv.project import forms
-from revolv.project.models import Category, Project, ProjectUpdate
+from revolv.project.models import Category, Project, ProjectUpdate, DonationLevel
 from revolv.tasks.sfdc import send_donation_info
 
+def stripe_callback(request, pk):
+    import stripe
+    stripe.api_key = settings.STRIPE_SECRET_KEY
+    token = stripe.Token.retrieve(request.POST['stripeToken'])
```
Comment (trawick) [inline revolv/project/views.py:27]: right now, token isn't used; keep this line and pass token to stripe.Charge.create down below

File: revolv/project/views.py [hunk 3]
```diff
@@ -14,12 +15,53 @@
...
-from revolv.payments.models import UserReinvestment
+from revolv.payments.models import UserReinvestment, Payment, PaymentType, Tip
 from revolv.payments.services import PaymentService
 from revolv.project import forms
-from revolv.project.models import Category, Project, ProjectUpdate
+from revolv.project.models import Category, Project, ProjectUpdate, DonationLevel
 from revolv.tasks.sfdc import send_donation_info
 
+def stripe_callback(request, pk):
+    import stripe
```
Comment (trawick) [inline revolv/project/views.py:25]: can this be moved to the third-party library import section above?

File: revolv/payments/models.py (hunk 1 above)
Comment (rlconley) [inline revolv/payments/models.py:549]: Yes, it was.

File: revolv/project/views.py [hunk 4]
```diff
@@ -14,12 +15,53 @@
...
+    stripe.api_key = settings.STRIPE_SECRET_KEY
```
Comment (trawick) [inline revolv/project/views.py:26]: I wonder if this should be done at global scope since it isn't per-request?

**Comment by rlconley** [comment]
Closed and re-created rebased on devel

**Pull request closed without merging**